src/tribol/mesh/MeshData.hpp

```diff
@@ -250,6 +265,9 @@ class MeshData {
      */
     TRIBOL_HOST_DEVICE const Array2DView<RealT>& getNodalNormals() const { return m_node_n; }
 
+    /// @overload
+    TRIBOL_HOST_DEVICE Array2DView<RealT>& getNodalNormals() { return m_node_n; }
```

Comment:
I'm having a little bit of a hard time tracking...I thought maybe the MeshData would have a nodal normal object. Then the mesh data can use the polymorphism to make sure the correct normal calculation is being invoked in a call to `Compute()`. Then, to get the nodal normal off the mesh data in this routine, would be getting it from the member nodal normal object. Let's discuss.

Reply:
Let's keep computation methods out of MeshData.  That's specific to a contact method, which the mesh shouldn't know or care about.